src/kindlemint/agents/market_research_agent.py

```
@@ -0,0 +1,702 @@
+            "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36"
+        }
+
+        for search_term in search_terms[:5]:  # Limit searches
+            try:
+                # Amazon search URL
+                search_url = (
+                    f"https://www.amazon.com/s?k={quote_plus(search_term)}&i=stripbooks"
+                )
+
+                async with aiohttp.ClientSession(headers=headers) as session:
+                    async with session.get(search_url) as response:
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)